pkg/objectfile/pool.go

```diff
@@ -234,7 +234,8 @@ func (p *Pool) NewFile(f *os.File) (_ *ObjectFile, err error) { //nolint:nonamed
 	// > Clients of ReadAt can execute parallel ReadAt calls on the same input source.
 	ef, err := elfNewFile(f)
 	if err != nil {
-		if errors.Is(err, &elf.FormatError{}) {
+		var elfErr *elf.FormatError
+		if errors.As(err, &elfErr) {
```

Comment:
Curious, what's the benefit of `error.As()` if `elfErr` is never used?

Reply:
Unfortunately errors.Is only works to check identical copies. Just doing errors.Is with an empty elf.FormatError is not equal to one that has the (unexported) fields set.

Reply:
Ah thank you, makes sense!